src/main/java/uk/gov/hmcts/darts/task/runner/impl/AudioLinkingAutomatedTask.java

```diff
@@ -0,0 +1,108 @@
+
+import jakarta.transaction.Transactional;
+import lombok.Getter;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.data.domain.Limit;
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.darts.common.entity.CourtCaseEntity;
+import uk.gov.hmcts.darts.common.entity.EventEntity;
+import uk.gov.hmcts.darts.common.entity.HearingEntity;
+import uk.gov.hmcts.darts.common.entity.MediaEntity;
+import uk.gov.hmcts.darts.common.entity.MediaLinkedCaseEntity;
+import uk.gov.hmcts.darts.common.repository.AutomatedTaskRepository;
+import uk.gov.hmcts.darts.common.repository.EventRepository;
+import uk.gov.hmcts.darts.common.repository.HearingRepository;
+import uk.gov.hmcts.darts.common.repository.MediaLinkedCaseRepository;
+import uk.gov.hmcts.darts.common.repository.MediaRepository;
+import uk.gov.hmcts.darts.event.enums.EventStatus;
+import uk.gov.hmcts.darts.log.api.LogApi;
+import uk.gov.hmcts.darts.task.api.AutomatedTaskName;
+import uk.gov.hmcts.darts.task.config.AutomatedTaskConfigurationProperties;
+import uk.gov.hmcts.darts.task.runner.AutoloadingManualTask;
+import uk.gov.hmcts.darts.task.service.LockService;
+
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+@Slf4j
+@Component
+public class AudioLinkingAutomatedTask extends AbstractLockableAutomatedTask
+    implements AutoloadingManualTask {
+
+    private final EventRepository eventRepository;
+    private final MediaRepository mediaRepository;
+    private final MediaLinkedCaseRepository mediaLinkedCaseRepository;
+
+    @Getter
+    private final Integer audioBufferSeconds;
+    private final HearingRepository hearingRepository;
+
+    protected AudioLinkingAutomatedTask(AutomatedTaskRepository automatedTaskRepository,
+                                        AutomatedTaskConfigurationProperties automatedTaskConfigurationProperties,
+                                        LogApi logApi, LockService lockService,
+                                        EventRepository eventRepository,
+                                        MediaRepository mediaRepository, MediaLinkedCaseRepository mediaLinkedCaseRepository,
+                                        @Value("${darts.automated-tasks.audio-linking.audio-buffer-seconds:0}")
+                                        Integer audioBufferSeconds, HearingRepository hearingRepository) {
+        super(automatedTaskRepository, automatedTaskConfigurationProperties, logApi, lockService);
+        this.eventRepository = eventRepository;
+        this.mediaRepository = mediaRepository;
+        this.mediaLinkedCaseRepository = mediaLinkedCaseRepository;
+        this.audioBufferSeconds = audioBufferSeconds;
+        this.hearingRepository = hearingRepository;
+    }
+
+    @Override
+    public AutomatedTaskName getAutomatedTaskName() {
+        return AutomatedTaskName.AUDIO_LINKING_TASK_NAME;
+    }
+
+    @Override
+    @Transactional
```

Comment:
is this going to make mean that the whole job one big transaction?

Reply:
This treats it as a spring transaction, as we are not doing any database locking with the @Lock annotation

Reply:
please see other comment about 1 failure not stopping the batch.

Reply:
If we remove the transaction one failure will still stop the batch, I will wrap some error handling around the processEvent logic

Reply:
I think there may be a scenario where we have a problem with the way this has been implemented:

     - Audio is received at 8:10pm
     - Event 1 is received 8:11pm
     - This audio linking job kicks in at 8:17pm, picking up event 1 along with 7500 other events
     - Versioned event (event 2) for event 1 received at 8:18pm. This will set the is_current flag to false for event 1
     
If it took a while to iterate through your loop of 7500 events, you are potentially locking the event and blocking the event from getting processed above. I agree with Chris it is safer if we do this using a 'save'. This is not a time crtitical automated task, so I don't think it matters if this adds on some processing time for this task to run.

Reply:
updated

Reply:
We might need to have a discussion to walk through this in abit more detail. We need to make sure that if we end up in the scenario detailed above that we don't end up overwriting changes from an update made whilst the task was looping through.

Reply:
Actioned updates following this call